Userland/Shell/Builtin.cpp

```diff
@@ -1090,16 +1090,51 @@ ErrorOr<int> Shell::builtin_time(Main::Arguments arguments)
 ErrorOr<int> Shell::builtin_umask(Main::Arguments arguments)
 {
     StringView mask_text;
+    bool symbolic_output = false;
 
     Core::ArgsParser parser;
+
+    parser.add_option(symbolic_output, "Produce symbolic output", "symbolic", 'S');
     parser.add_positional_argument(mask_text, "New mask (omit to get current mask)", "octal-mask", Core::ArgsParser::Required::No);
 
     if (!parser.parse(arguments, Core::ArgsParser::FailureBehavior::PrintUsage))
         return 1;
 
+    auto parse_symbolic_digit = [](int digit) -> ErrorOr<String> {
+        StringBuilder builder;
+
+        if ((digit & 4) == 0)
```

Comment:
Am I crazy or are these checks inverted?

0644 means `rw-r--r--`

But with this code, we'll shift it over 6, for user, and check:

- (6 & 4 == 0) ?  nope, it's 4. '-'
- (6 & 2 == 0)? nope, it's 2. '-'
- (6 & 1 == 0? yep, ok,  'x'

And all the reported permissions will be backwards. `--xrw-rw-`

Reply:
I don't think so. WSL Ubuntu has the same output?

| WSL | Serenity |
|-----|----------|
|![image](https://github.com/SerenityOS/serenity/assets/36214028/6513eee1-8cf9-402f-8e3e-bf7b3438ad05) | ![image](https://github.com/SerenityOS/serenity/assets/36214028/af1962db-dc47-4b7c-b00c-ee110d8d22c0) |